akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/AccountExample2.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.akka.persistence.typed;
+
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.persistence.typed.PersistenceId;
+import akka.persistence.typed.javadsl.*;
+
+public class AccountExample2 extends PersistentBehavior<AccountExample2.AccountCommand, AccountExample2.AccountEvent, AccountExample2.Account> {
+
+  interface AccountCommand {}
+  public static class CreateAccount implements AccountCommand {}
+  public static class Deposit implements AccountCommand {
+    public final double amount;
+
+    public Deposit(double amount) {
+      this.amount = amount;
+    }
+  }
+  public static class Withdraw implements AccountCommand {
+    public final double amount;
+
+    public Withdraw(double amount) {
+      this.amount = amount;
+    }
+  }
+  public static class CloseAccount implements AccountCommand {}
+
+  interface AccountEvent {
+    Account applyTo(Account state);
+  }
+  public static class AccountCreated implements AccountEvent {
+    @Override
+    public Account applyTo(Account state) {
+      return state.created(this);
+    }
+  }
+  public static class Deposited implements AccountEvent {
+    public final double amount;
+
+    Deposited(double amount) {
+      this.amount = amount;
+    }
+
+    @Override
+    public Account applyTo(Account state) {
+      return state.deposited(this);
+    }
+  }
+  public static class Withdrawn implements AccountEvent {
+    public final double amount;
+
+    Withdrawn(double amount) {
+      this.amount = amount;
+    }
+
+    @Override
+    public Account applyTo(Account state) {
+      return state.withdrawn(this);
+    }
+  }
+  public static class AccountClosed implements AccountEvent {
+    @Override
+    public Account applyTo(Account state) {
+      return state.closed(this);
+    }
+  }
+
+  interface Account {
+    default Account created(AccountCreated event) {
+      throw new IllegalStateException();
```

Comment:
i know this is an OO pattern, but it's pretty weird to define all possible operations on the domain object although it only supports some of them in the concrete states

Reply:
Maybe if these methods were split into a separate interface called `AccountEventHandler` it would seem less weird?